### PR TITLE
Documentation

### LINKED ISSUE #697: Document our dev workflow/test driven development

### PR DESCRIPTION
**Pull request by maxneuvians**
Closes #697

### CODE REVIEW

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -26,6 +26,40 @@ The advantages here are significant: for one the loading experience is faster as
...
 
+### Test Driven Development
+
+Developers on this project follow a process that is loosely based on [Test Driven Development](https://en.wikipedia.org/wiki/Test-driven_development "Test Driven Developement") and other practices associated with "small a" [Agile](https://en.wikipedia.org/wiki/Agile_software_development "Agile") development. Without following strict guidelines, developers agreed that an iterative and continous approach is preferable to segmented release cycles.
+
+To facilitate this process, we have chosen to use a [Continous Integration](https://en.wikipedia.org/wiki/Continuous_integration "Continous Integration") and [Continouse Delivery](https://en.wikipedia.org/wiki/Continuous_delivery "Continouse Delivery") process whereby new features and code are added to the product frequenty, reviewed and verified, and shipped as soon as they are done. We are currently using [https://circleci.com/](https://circleci.com/ "Circle CI") as our platform for running all our unit tests before deploying. Alongside that we are also using [https://heroku.com](https://heroku.com "Heroku") to create staging versions of our application for manual review.
```
Comment (sastels) [inline DOCUMENTATION.md]: typo "Continouse Delivery"

File: DOCUMENTATION.md [hunk 2]
```diff
@@ -26,6 +26,40 @@ The advantages here are significant: for one the loading experience is faster as
...
+4. Features are broken down into issues on GitHub and developers agree on how much time each issue potentially can take
+5. Developers self-assign issues from the list of open issues in GitHub.
+6. To resolve an issue a developer:
+   - Checks out a new branch from `master`
+   - Builds the feature or resolves the bug described in the issue
+   - Continously commits code to their branch at sensible intervals
+   - Writes unit and integration tests for the code they have added
+   - Changes/removes unit and integration tests for code they have modified
+   - Runs the test suite that ensure all tests pass
+   - Uses a code coverage tool to ensure all signifcant logic has been tested
```
Comment (sastels) [inline DOCUMENTATION.md]: typo signifcant

File: DOCUMENTATION.md [hunk 3]
```diff
@@ -26,6 +26,40 @@ The advantages here are significant: for one the loading experience is faster as
...
+   - Commit the final code, pushes the code to GitHub, and opens a pull request in GitHub
```
Comment (sastels) [inline DOCUMENTATION.md]: should be "Commits"

File: DOCUMENTATION.md [hunk 4]
```diff
@@ -26,6 +26,40 @@ The advantages here are significant: for one the loading experience is faster as
...
+   - Awaits CircleCI to verify that all our tests will pass
+   - Spot check features using Heroku review apps
+   - Requests other members of the team to review the pull request
+   - If other team members have comments, they will note these in GitHub for the developer to either fix them or discuss
+   - Other team members approve the pull request once all open discussion has been resolved
```
Comment (sastels) [inline DOCUMENTATION.md]: could note that we only require one approval.

File: DOCUMENTATION.md [hunk 5]
```diff
@@ -26,6 +26,40 @@ The advantages here are significant: for one the loading experience is faster as
...
+   - Pull requests are then closed and their associated branch merged into master
```
Comment (sastels) [inline DOCUMENTATION.md]: Might be better to just say that pull requests are merged into master? Just that in GitHub we click the pull request's "merge" button, not the "close".

File: DOCUMENTATION.md [hunk 6]
```diff
@@ -26,6 +26,40 @@ The advantages here are significant: for one the loading experience is faster as
...
+   - CircleCI verifies once again that all tests pass and deploys the code into production
```
Comment (sastels) [inline DOCUMENTATION.md]: maybe "all tests pass on the updated master..."

**Review by sastels — APPROVED** [review]
minor comments, 🦑 🦑

**Comment by maxneuvians** [comment]
@sastels thank you for feedback - much appreciated!

**Pull request merged**